src/test/java/org/springframework/data/r2dbc/repository/query/PartTreeR2dbcQueryUnitTests.java

```diff
@@ -684,6 +687,59 @@ private RelationalParametersParameterAccessor getAccessor(R2dbcQueryMethod query
 		return new RelationalParametersParameterAccessor(queryMethod, values);
 	}
 
+	private static String formatOperation(PreparedOperation<?> preparedOperation){
+		return formatQuery(preparedOperation.get());
+	}
+
+	private static String formatQuery(String query){
+		String firstKeyword = "SELECT";
+		String lastKeyword = "FROM";
+		
+		int indexOfFirstKeyWord = query.toUpperCase().indexOf(firstKeyword);
+		int indexOfLastKeyWord = query.toUpperCase().indexOf(lastKeyword);
+		
+		if(indexOfFirstKeyWord!=0 || indexOfFirstKeyWord>=indexOfLastKeyWord){
+			return query;
+		}
+		
+		String fields = query.substring(firstKeyword.length(), indexOfLastKeyWord);
+		String sortedFields = sortFields(fields);
+		
+		StringBuilder formattedQuery = new StringBuilder();
+		formattedQuery.append(firstKeyword);
+		formattedQuery.append(" ");
+		formattedQuery.append(sortedFields);
+		formattedQuery.append(" ");
+		formattedQuery.append(query.substring(indexOfLastKeyWord, query.length()));
+		
+		return formattedQuery.toString();
+        }
+
+        private static String sortFields(String fields){
+		List<String> sortedFieldsList = new ArrayList<>();
+		StringBuilder fieldBuilder = new StringBuilder();
+		StringBuilder sortedFields = new StringBuilder();
+		
+		for(int i=0;i<fields.length();i++){
+			if(fields.charAt(i)==','){
+				sortedFieldsList.add(fieldBuilder.toString().trim());
+				fieldBuilder = new StringBuilder();
```

Comment:
```suggestion
				fieldBuilder.setLength(0);
```

should have same effect but fewer allocations.